scipy/stats/_continuous_distns.py

```diff
@@ -3456,8 +3456,25 @@ def _munp(self, n, a, c):
         return sc.poch(a, n*1.0/c)
 
     def _entropy(self, a, c):
-        val = sc.psi(a)
-        return a*(1-val) + 1.0/c*val + sc.gammaln(a) - np.log(abs(c))
+        def regular(a, c):
+            val = sc.psi(a)
+            A = a * (1 - val) + 1.0 / (c * val)
```

Comment:
```suggestion
            A = a * (1 - val) + 1.0 / c * val
```

This should fix the test failures.

Reply:
The way it's currently written is way too confusing. I'm changing it to `val / c` instead.